components/button/test/button-icon.visual-diff.js

```diff
@@ -21,100 +21,25 @@ describe('d2l-button-icon', function() {
-			await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
-		});
-
-		it('focus', async function() {
-			await focus(page, '#normal');
-			const rect = await visualDiff.getRect(page, '#normal');
-			await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
-		});
-
-		it('disabled', async function() {
-			const rect = await visualDiff.getRect(page, '#disabled');
-			await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
-		});
-
-	});
-
-	describe('translucent', function() {
-
-		it('normal', async function() {
-			const rect = await visualDiff.getRect(page, '#translucent-enabled');
-			await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
-		});
-
-		it('focus', async function() {
-			await focus(page, '#translucent-enabled > d2l-button-icon');
-			const rect = await visualDiff.getRect(page, '#translucent-enabled');
-			await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
-		});
-
-		it('hover', async function() {
-			await hover(page, '#translucent-enabled > d2l-button-icon');
-			const rect = await visualDiff.getRect(page, '#translucent-enabled');
-			await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
-		});
-
-		it('disabled', async function() {
-			const rect = await visualDiff.getRect(page, '#translucent-disabled');
-			await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
-		});
-
-	});
-
-	describe('custom', function() {
-
-		it('normal', async function() {
-			const rect = await visualDiff.getRect(page, '#custom');
-			await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
-		});
-
-		it('hover', async function() {
-			await page.hover('#custom');
-			const rect = await visualDiff.getRect(page, '#custom');
-			await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
-		});
-
-		it('focus', async function() {
-			await focus(page, '#custom');
-			const rect = await visualDiff.getRect(page, '#custom');
-			await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
-		});
-
-	});
-
-	const hover = (page, selector) => {
-		const p = page.evaluate((selector) => {
-			return new Promise((resolve) => {
-				const elem = document.querySelector(selector);
-				elem.shadowRoot.querySelector('button').addEventListener('transitionend', (e) => {
-					if (e.propertyName === 'background-color') resolve();
+	[
+		{category: 'normal', tests: ['normal', 'hover', 'focus', 'disabled']},
+		{category: 'primary', tests: ['normal', 'hover', 'focus', 'primary-disabled']},
+	].forEach((entry) => {
+		describe(entry.category, () => {
+			entry.tests.forEach((name) => {
+				it(name, async function() {
+					if (name === 'hover') {
+						await page.hover(`#${entry.category}`);
+					} else if (name === 'focus') {
+						await focus(page, `#${entry.category}`);
```

Comment:
You'll need the `focus` method still for this line.

Reply:
looks like I've put this in the wrong file, though I might as well get all of the buttons running test like this. 
(sorry, I've only now got test working locally. I'll make sure to run them, prior to pushing commits up, in the future.)

Reply:
Haha yeah I was going to mention that but wasn't sure if you were finished. Up to you whether you want to tackle them all now or we can do it as a follow-up.